### PR TITLE
remove duplicate log fields

### PR DESCRIPTION
**Pull request by jaypipes**
The use of `logr.Logger.WithValues` in the
`pkg/runtime/log.AdoptResource` function was not actually overwriting
previously set key/value pairs -- for instance, for the resource's kind,
generation or namespace. Instead, these structured log record fields
were simply being appended to the end of the `[]interface{}` that
`logr.Logger` uses to track log record fields.

See
https://pkg.go.dev/github.com/go-logr/logr#readme-why-key-value-pairs-and-not-a-map
for information on why a `[]interface{}` is used instead of a
`map[string]interface{}`, which would have avoided this bug/problem.

This patch removes the use of `logr.Logger.WithValues` in the
`ResourceLogger.Debug`, `ResourceLogger.Enter|Exit` and
ResourceLogger.Info` methods and replaces with direct passing of the
core resource fields into the `logr.Logger.Info` method.

Fixes Issue aws-controllers-k8s/runtime#938

By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by vijtrip2** [comment]
>This patch removes the use of logr.Logger.WithValues and replaces with
direct passing of the core resource fields into the logr.Logger.Info
method.

What am i missing here? I still see "log.WithValues(vals...)".
I only see code being refactored into separate method. Apologies if I had an oversight.

**Comment by jaypipes** [comment]
> > This patch removes the use of logr.Logger.WithValues and replaces with
> > direct passing of the core resource fields into the logr.Logger.Info
> > method.
> 
> What am i missing here? I still see "log.WithValues(vals...)".
> I only see code being refactored into separate method. Apologies if I had an oversight.

Ugh, sorry :( Serves me right for trying to rush in a PR before a PE review... will fix shortly.

**Comment by jaypipes** [comment]
@vijtrip2 thanks for the catch. fixed.

File: pkg/runtime/log/resource.go [hunk 1]
```diff
@@ -163,28 +150,38 @@ func InfoResource(
...
 	}
 	if len(additionalValues) > 0 {
 		vals = append(vals, additionalValues...)
 	}
+	return vals
+}
+
+// AdaptAdoptedResource returns a logger with log values set for the adopted
+// resource's kind, namespace, name, etc
+func AdaptAdoptedResource(
```
Comment (vijtrip2) [inline pkg/runtime/log/resource.go:179]: femto nit: Maybe these logging methods can be removed now since we log directly now.
Can be done in future PRs.

**Review by vijtrip2 — APPROVED** [review]

**Pull request merged**